lib/models/project.js

```diff
@@ -529,7 +529,8 @@ Project.prototype.findAddonByName = function(name) {
   }
 
   return find(this.addons, function(addon) {
-    return name.indexOf(addon.name) > -1 || (addon.pkg && name.indexOf(addon.pkg.name) > -1);
+    return (name.slice(addon.name && -addon.name.length) === addon.name) ||
```

Comment:
There is a [test](https://github.com/ember-cli/ember-cli/blob/master/tests/unit/commands/install-test.js#L158) that causes `findAddonByName('ember-cli/ember-cli-qunit')` to be run.